openmc/tallies.py

```diff
@@ -1848,6 +1907,116 @@ def swap_filters(self, filter1, filter2, inplace=False):
+            is populated with data by the StatePoint.read_results() method.
+
+        """
+
+        # Check that results have been read
+        if not self.derived and self.sum is None:
+            msg = 'Unable to use tally arithmetic with Tally ID="{0}" ' \
+                  'since it does not contain any results.'.format(self.id)
+            raise ValueError(msg)
+
+        # Swap the nuclides in the Tally
+        nuclide1_index = self.nuclides.index(nuclide1)
+        nuclide2_index = self.nuclides.index(nuclide2)
+        self.nuclides[nuclide1_index] = nuclide2
+        self.nuclides[nuclide2_index] = nuclide1
+
+        # Copy the tally data
+        self_mean = copy.deepcopy(self.mean)
+        self_std_dev = copy.deepcopy(self.std_dev)
+
+        # Swap nuclide 1 in place of nuclide 2
+        self._mean = np.delete(self.mean, nuclide2_index, axis=1)
+        self._mean = np.insert(self.mean, nuclide2_index,
+                               self_mean[:,nuclide1_index,:], axis=1)
+        self._std_dev = np.delete(self.std_dev, nuclide2_index, axis=1)
+        self._std_dev = np.insert(self.std_dev, nuclide2_index,
+                                  self_mean[:,nuclide1_index,:], axis=1)
+
+        # Swap nuclide 2 in place of nuclide 1
+        self._mean = np.delete(self.mean, nuclide1_index, axis=1)
+        self._mean = np.insert(self.mean, nuclide1_index,
+                               self_mean[:,nuclide2_index,:], axis=1)
+        self._std_dev = np.delete(self.std_dev, nuclide1_index, axis=1)
+        self._std_dev = np.insert(self.std_dev, nuclide1_index,
+                                  self_mean[:,nuclide2_index,:], axis=1)
+
+    def swap_scores(self, score1, score2):
+        """Reverse the ordering of two scores in this tally
+
+        This is a helper method for tally arithmetic which helps align the data
+        in two tallies with shared scores. This method copies reverses the order
+        of the two scores in place.
+
+        Parameters
+        ----------
+        score1 : Score
+            The score to swap with score2
+
+        score2 : Score
+            The score to swap with score1
+
+        Raises
+        ------
+        ValueError
+            If this is a derived tally or this method is called before the tally
+            is populated with data by the StatePoint.read_results() method.
+
+        """
+
+        # Check that results have been read
+        if not self.derived and self.sum is None:
+            msg = 'Unable to use tally arithmetic with Tally ID="{0}" ' \
+                  'since it does not contain any results.'.format(self.id)
+            raise ValueError(msg)
+
+        # Swap the scores in the Tally
+        score1_index = self.scores.index(score1)
+        score2_index = self.scores.index(score2)
+        self.scores[score1_index] = score2
+        self.scores[score2_index] = score1
+
+        # Copy the tally data
+        self_mean = copy.deepcopy(self.mean)
```

Comment:
Do you need to deepcopy the entire arrays, or just the portions corresponding to scores 1 and 2?